### PR TITLE
Implemented logic for publishing messages to SNS

### PR DESCRIPTION
**Pull request by jzoldak**
@benpatterson here's a preview. Hoping to get this wrapped up, I still want to change some of the nomenclature, e.g. I'm not sure I like the word "Trigger" or references to jenkins where they should be more generic, etc.

### CODE REVIEW

File: testeng_triggers/testeng_triggers.py [hunk 1]
```diff
@@ -1,46 +1,80 @@
...
+PROVISIONING_TOPIC = os.environ.get('PROVISIONING_TOPIC', 'insert_sns_arn_here')
+SITESPEED_TOPIC = os.environ.get('SITESPEED_TOPIC', 'insert_sns_arn_here')
+
+
+class TriggerHttpRequestHandler(BaseHTTPRequestHandler, object):
     """
-    Handler for the stub HTTP service.
+    Handler for the HTTP service.
     """
     protocol = "HTTP/1.0"
```
Comment (benpatterson) [inline testeng_triggers/testeng_triggers.py]: Just curious if HTTP 1.1 is an option...

File: testeng_triggers/testeng_triggers.py [hunk 2]
```diff
@@ -1,46 +1,80 @@
...
+        repo = data.get('repository')
+        repo_name = repo.get('full_name')
+        if repo_name != HANDLED_REPO:
+            # We only want to take action on a specific repo, so
+            # even if another repo gets configured to send webhooks
+            # to this app send back a 200 to GitHub
+            LOGGER.debug('Unhandled repo: {}'.format(repo_name))
+            return None
+
+        msg_id = None
```
Comment (benpatterson) [inline testeng_triggers/testeng_triggers.py]: nit: move this below the comment

Comment (benpatterson) [inline testeng_triggers/testeng_triggers.py]: Oh I see this now....I take back my previous comment!

File: testeng_triggers/testeng_triggers.py [hunk 3]
```diff
@@ -50,143 +84,72 @@ def request_content(self):
         try:
             length = int(self.headers.getheader('content-length'))
 
-        except (TypeError, ValueError):
+        except (TypeError, ValueError):  # pragma: no cover
             return ""
```
Comment (benpatterson) [inline testeng_triggers/testeng_triggers.py]: Log message here as well?

**Comment by benpatterson** [comment]
Still reading through this, but I'd like to hear more about the no-cover sets on the various exceptions. Perhaps we can discuss offline.

File: testeng_triggers/test/test_server.py [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+        """
+        """
+        address = ('0.0.0.0', 0)
+        HTTPServer.__init__(self, address, TriggerHttpRequestHandler)
+
+        server_thread = threading.Thread(target=self.serve_forever)
+        server_thread.daemon = True
+        server_thread.start()
+
+    def shutdown(self):
```
Comment (benpatterson) [inline testeng_triggers/test/test_server.py]: `__exit__` ?

File: testeng_triggers/test/test_server.py [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+
+class TriggerServerTestCase(TestCase):
+    """TestCase class for verifying the HTTP server that
+    is servicing the webhooks from GitHub.
+    """
+    def setUp(self):
+        """These tests start the server to test it. """
+        super(TriggerServerTestCase, self).setUp()
+        self.server = ThreadedHTTPServer()
+        self.addCleanup(self.server.shutdown)
```
Comment (benpatterson) [inline testeng_triggers/test/test_server.py]: Gotcha. I suppose that makes my `__exit__` comment more of a nit.

**Comment by jzoldak** [comment]
@benpatterson the no cover pragma is just for test coverage. See http://coverage.readthedocs.org/en/latest/excluding.html

**Comment by jzoldak** [comment]
Closing in favor of https://github.com/jzoldak/build-pipeline/pull/1

**Pull request closed without merging**